dev/statement_serializer.h

```diff
@@ -1140,12 +1150,17 @@ namespace sqlite_orm {
 
                 std::stringstream ss;
                 ss << streaming_identifier(column.name);
-                if(!context.skip_types_and_constraints) {
+                if(!context.skip_types_and_constraints_except_unindexed) {
                     ss << " " << type_printer<field_type_t<column_type>>().print();
                     ss << streaming_column_constraints(
                         call_as_template_base<column_constraints>(polyfill::identity{})(column),
                         column.is_not_null(),
                         context);
+                } else {
+                    using constraints_tuple = typename column_type::constraints_type;
```

Comment:
@trueqbit probably this algorithm can be implemented way more beautiful. Do you have any idea?

Reply:
Yes, indeed ;)
```c++
        template<class G, class S, class... Op>
        struct statement_serializer<column_t<G, S, Op...>, void> {
            using statement_type = column_t<G, S, Op...>;

            template<class Ctx>
            std::string operator()(const statement_type& column, const Ctx& context) const {
                std::stringstream ss;
                ss << streaming_identifier(column.name);
                if(!context.fts5_columns) {
                    ss << " " << type_printer<field_type_t<column_field<G, S>>>().print();
                }
                ss << streaming_column_constraints(
                    call_as_template_base<column_constraints>(polyfill::identity{})(column),
                    column.is_not_null(),
                    context);
                return ss.str();
            }
        };
```

And the accompanying `streaming_column_constraints` in _serializing_util.h_:
```c++
        template<class... Op, class Ctx>
        std::ostream& operator<<(std::ostream& ss,
                                 std::tuple<const streaming<stream_as::column_constraints>&,
                                            const column_constraints<Op...>&,
                                            const bool&,
                                            Ctx> tpl) {
            const auto& column = std::get<1>(tpl);
            const bool& isNotNull = std::get<2>(tpl);
            auto& context = std::get<3>(tpl);

            using constraints_tuple = decltype(column.constraints);
            if(!context.fts5_columns) {
                iterate_tuple(column.constraints, [&ss, &context](auto& constraint) {
                    ss << ' ' << serialize(constraint, context);
                });
                constexpr bool hasExplicitNullableConstraint =
                    mpl::invoke_t<mpl::disjunction<check_if_has_type<null_t>, check_if_has_type<not_null_t>>,
                                  constraints_tuple>::value;
                if SQLITE_ORM_CONSTEXPR_IF(!hasExplicitNullableConstraint) {
                    if(isNotNull) {
                        ss << " NOT NULL";
                    } else {
                        ss << " NULL";
                    }
                }
            } else {
                constexpr bool hasUnindexedOption =
                    mpl::invoke_t<check_if_has_type<unindexed_t>, constraints_tuple>::value;
                if SQLITE_ORM_CONSTEXPR_IF(hasUnindexedOption) {
                    ss << " UNINDEXED";
                }
            }

            return ss;
        }
```

Reply:
thanks